dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,635 @@
+`allow_no_crop` is `False` and `thresholds` does not contain `0.0`, it is good to increase the
+`num_attempts` value as otherwise it may loop for a very long time.
+
+Inputs: 0: `bboxes`, (1:`labels`, )
+
+First input `bboxes` refers to bounding boxes that are provided as a 2-dimensional tensor where the
+first dimension refers to the index of the bounding box and the second dimension refers to the
+index of the coordinate.
+Coordinates are relative to the original image dimensions (i.e. range [0.0, 1.0]), representing
+either `start` and `end` of the region (e.g. `[left, top, right, bottom]`, or `ltrb`) or `start` and
+`shape` (e.g. `[left, top, width, height]` or `xywh`). In the case of volumetric inputs, bounding
+boxes are encoded as `[left, top, front, right, bottom, back]` or
+`[left, top, front, width, height, depth]`.
+
+A second input `labels` can be optionally provided, representing the labels associated with each of
+the bounding boxes.
+
+Outputs: 0:`anchor`, 1:`shape`, 2:`bboxes`, (3:`labels`,)
+
+The result of the prospective crop operation is provided as two separate outputs `anchor` and
+`shape`, that can be feed directly to `Slice` operator to perform the actual crop operation to the
+original image. `anchor` and `shape` contains starting coordinates and dimensions for the crop, as
+`[x, y, (z,)]` and `[w, h, (d,)]` respectively. The coordinates can be represented in absolute or
+relative terms, depending of whether we used a fixed `crop_shape` or not.
+
+Third and fourth outputs correspond to the adjusted bounding boxes and optionally
+their corresponding labels. Bounding boxes are always specified in relative coordinates.)code")
+    .NumInput(1, 2)  // [boxes, labels (optional),]
+    .NumOutput(3)    // [anchor, shape, bboxes, labels (optional),]
+    .AdditionalOutputsFn(
+      [](const OpSpec& spec) {
+        return spec.NumInput() - 1;  // +1 if labels are provided
+      })
     .AddOptionalArg(
         "thresholds",
-        R"code(Minimum overlap (Intersection over union) of the bounding boxes with respect to the prospective crop.
-Selected at random for every sample from provided values. Default imposes no restrictions on Intersection over Union for boxes and crop.)code",
+        R"code(Minimum intersection-over-union (IoU) of the bounding boxes with respect to the
+prospective crop. Each sample will select one of the `thresholds` randomly, and the operator will
+try a given number of attempts (see `num_attempts`) to produce a random crop window that yields an
+IoU above the selected threshold.)code",
         std::vector<float>{0.f})
     .AddOptionalArg(
         "aspect_ratio",
-        R"code(Range `[min, max]` of valid aspect ratio values for new crops. Value for `min` should be greater or equal to `0.0`.
-Default values disallow changes in aspect ratio.)code",
+        R"code(Range `[min, max]` of valid aspect ratio values for new crops.
```

Comment:
For 3D, we should either have additional arguments or put 6 values here, so we can have distinct X/Y, X/Z and Y/Z ratio ranges.